tests/test_attr_datatypes.py

```diff
@@ -79,10 +79,14 @@ def test_numeric_array_attr_datatypes():
         assert hfile.attrs['uint64_array'].dtype == np.dtype('>u8')
         assert hfile.attrs['float32_array'].dtype == np.dtype('<f4')
 
-        assert hfile.attrs['vlen_str_array'][0] == b'Hello'
-        assert hfile.attrs['vlen_str_array'][1] == b'World!'
 
-        assert hfile.attrs['vlen_str_array'].dtype == np.dtype('S6')
+def test_string_array_attr_datatypes():
+    with pyfive.File(ATTR_DATATYPES_HDF5_FILE) as hfile:
+        assert hfile.attrs['vlen_str_array'][0] == 'Hello'
+        assert hfile.attrs['vlen_str_array'][1] == 'World!'
+
+        assert hfile.attrs['vlen_str_array'].dtype == np.dtype('O')
+        assert hfile.attrs['vlen_str_array'].dtype.metadata == {'h5py_encoding': 'utf-8'}
```

Comment:
This reads back now as strings. Because I had to fix this, I've taken the chance to move this to its own test function.

Reply:
So, I think this is a different behaviour from h5py. 
 ```
with h5py.File(ATTR_DATATYPES_HDF5_FILE) as hfile:
    ...:     x=hfile.attrs['vlen_str_array'][0]
    ...:     print(x)
b'Hello'
```
so I'm thinking this is not what we want?

Reply:
I'll double check this one.

Reply:
And from looking at this, the old version doesn't seem to be variable length at all.